aiosmtpd/controller.py

```diff
@@ -21,7 +21,7 @@ def __init__(self, handler, loop=None, hostname='::0', port=8025,
         self.ready_timeout = os.getenv(
             'AIOSMTPD_CONTROLLER_TIMEOUT', ready_timeout)
         # For exiting the loop.
-        self._rsock, self._wsock = socket.socketpair()
+        self._rsock, self._wsock = self.loop._socketpair()
```

Comment:
We can't use `loop._socketpair()` here since that's a non-public API, meaning it could change out from underneath us at any time.  The problem with going back to `socket.socketpair()` is that that API does not exist on Python 3.4.

I think we should just use the sample code [given here](https://docs.python.org/3/library/asyncio-protocol.html?highlight=socketpair#register-an-open-socket-to-wait-for-data-using-a-protocol).